src/components/Profile/Profile.jsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { Routes, Route, Navigate, useLocation } from "react-router-dom";
+import Sidebar from "../Sidebar/Sidebar";
+import Comment from "../Comment/Comment";
+import Form from "../Form/Form";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
+import useFormAndValidation from "../../hooks/useFormAndValidation";
+import "./Profile.css";
+
+const Profile = ({
+  userComments,
+  handleLogout,
+  isLoading,
+  handleSubmit,
+  handleDelete,
+  handleCommentToEdit,
+}) => {
+  const currentUser = React.useContext(CurrentUserContext);
+  const [isUpdateActive, setIsUpdateActive] = React.useState(false);
+  const currentLocation = useLocation();
+  const {
+    values,
+    setValues,
+    handleChange,
+    errors,
+    resetForm,
+    isValid,
+    isButtonDisabled,
+    setIsButtonDisabled,
+  } = useFormAndValidation();
+
+  const handleFormSubmit = () => {
+    handleSubmit(values)
+      .then(() => {
+        setIsUpdateActive(false);
+        resetForm();
+      })
+      .catch(console.error);
+  };
+
+  React.useEffect(() => {
+    resetForm();
+    setValues({ name: currentUser.name, email: currentUser.email });
+    setIsUpdateActive(false);
+  }, [currentUser, currentLocation]);
+
+  React.useEffect(() => {
+    if (
+      values.name === currentUser.name &&
+      values.email === currentUser.email
+    ) {
+      setIsButtonDisabled(true);
+    } else if (isValid) {
+      setIsButtonDisabled(false);
+    }
+  }, [values]);
+
+  return (
+    <div className="profile">
+      <Sidebar handleLogout={handleLogout} />
+      <Routes>
+        <Route path="/" element={<Navigate to="info" replace />} />
+        <Route
+          path="info"
+          element={
+            <Form handleSubmit={handleFormSubmit} values={values}>
+              <label
+                className={`form__label ${errors.name ? "form__error" : ""}`}
+                htmlFor="update-name"
+              >
+                Username*
+                <input
+                  className="form__input form__input_profile"
+                  type="text"
+                  name="name"
+                  id="update-name"
+                  value={values.name || ""}
```

Comment:
Good job that you don't forget about `value` in the inputs if you have `onChange`